### ISSUE TITLE
Which method should I use?

### ISSUE THREAD

**Issue by buzem** [stellargraph/stellargraph#1884]
Hi,

I am trying to predict the type of links(edge) between nodes in my graph.There is only two types of edges : positive or negative. I tried to adapt the ComplEx Tutorial into my dataset however I couldn't find a way how I can get the ROC_AUC score ? In tutorial, they evaluated the results only based on MRR score with the help of rank_edges_against_all_nodes.

I'd be glad if you can help

**Comment by huonw** [comment]
Hi @buzem, thanks for getting in touch. I think the most direct way to do this is something along the lines of:

1. train your model (sounds like you've got this going well 👍 )
2. use the trained model to compute predictions on your whole labelled test/validation/hold-out set
3. compare the predicted labels against the true labels with a function like https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html

Another way to do this would be to add [an `AUC` metric](https://www.tensorflow.org/api_docs/python/tf/keras/metrics/AUC) when you compile your keras `Model` (like `model.compile(..., metrics=AUC(...))`), but I've never done this personally.

Does that makes sense?

(Also, I imagine you've selected the ComplEx model because you have nodes without any associated node features/data? Just in case you haven't seen it, there's [a table of other algorithms available for link prediction in different situations](https://stellargraph.readthedocs.io/en/stable/demos/link-prediction/index.html), most of those demos are doing binary (positive/negative) classification too.)
